### PR TITLE
fix: Handle invalid well values in SoftmaxPro

### PR DESCRIPTION
**Pull request by nathan-stender**
Populate invalid values (e.g. "Path?" from provided example) with "NaN"

### CODE REVIEW

File: src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py [hunk 1]
```diff
@@ -291,7 +292,7 @@ class DataElement:
     temperature: Optional[float]
     wavelength: float
     position: str
-    value: float
+    value: float | InvalidJsonFloat
```
Comment (bdworth) [inline src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py:295]: `JsonFloat` is an alias for `float | InvalidJsonFloat`

File: src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py [hunk 2]
```diff
@@ -331,10 +332,7 @@ def create(
                     temperature=temperature,
                     wavelength=wavelength,
                     position=str(position),
-                    value=try_float(
-                        value,
-                        f"value from block plate {plate_name} and wavelength {wavelength}",
-                    ),
+                    value=try_float_or_none(value) or InvalidJsonFloat.NaN,
```
Comment (bdworth) [inline src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py:335]: This is fine, but anticipating that we might want to do more of this, how about adding a utility method a la the following and calling it here?
```
def try_float_or_nan(value: Optional[str]) -> JsonFloat:
    if not value:
        return InvalidJsonFloat.NaN
    try:
        return float(value)
    except ValueError:
        return InvalidJsonFloat.NaN
```

**Review by bdworth — APPROVED** [review]
fine as is, but two suggestions

File: src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py (hunk 2 above)
Comment (nathan-stender) [reply src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py:335]: There are some other spots that use this already, plan to follow up with a util.

File: src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py (hunk 1 above)
Comment (nathan-stender) [reply src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py:295]: Will get this in util PR as well

**Pull request merged**